src/internal/connector/graph/errors.go

```diff
@@ -0,0 +1,92 @@
+package graph
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/microsoftgraph/msgraph-sdk-go/models/odataerrors"
+	"github.com/pkg/errors"
+)
+
+// ---------------------------------------------------------------------------
+// Error Interpretation Helpers
+// ---------------------------------------------------------------------------
+
+// ErrIs checks for graph specific errors.  Returns nil if the errors do not
+// match.  Returns errors.Wrap(graphErr, err) if the errors match.
+func ErrIs(err, graphErr error) error {
+	switch true {
+	// http timeouts, not retried by the graph client
+	case errors.Is(graphErr, ErrTimeout):
+		if isTimeoutErr(err) {
+			return errors.Wrap(ErrTimeout, err.Error())
```

Comment:
I'm not really a fan of using wrapping like this. I would suggest using the custom error package we have. This form of wrapping will lose any stack track information or other info that the error may have

Reply:
ah, totally forgot about that package.  Truth be told I'm not a fan of the using As for sentinel errors.  Kinda wish we had a `errors.Wrap(err, sentinalErr)`.  But it'll suffice.